content/articles/wavelet-transform-analysis-of-1d-signals-using-python/index.md

```diff
@@ -0,0 +1,221 @@
+```
+
+We plot the noisy signal `x_noisy` and the denoised signal `x-denoise` for visualisation.
+
+```python
+plt.figure(figsize=(20, 10), dpi=100)
+plt.plot(x_noisy)
+plt.plot(x_denoise)
+```
+
+When plotting, we first create a figure using the `plt.figure()` function. This figure is of the size 20x10 pixel as described by the function property `figsize`. Next, the noisy and the denoised signals are plotted in the exact figure using the `plt.plot()` function. The output of this plot is as shown below:
+
+![Plot of the signals](/engineering-education/wavelet-transform-analysis-of-1d-signals-using-python/wavelet-two.png)
+
+The blue signal is the noisy ecg signal, and the red is the denoised signal. Let us now look at how to denoise the audio signal since it is also a sample of a 1-D signal.
+
+### Python code for denoising an audio signal
+The first step is importing the libraries that we are going to use.
+
+```Python
+from scipy.io import wavfile
+import numpy as np
+from skimage.restoration import denoise_wavelet
+import matplotlib.pyplot as plt
+```
+
+We are importing the `wavfile` from the `scipy.io`. It is used to read audio signals in the `.wav` format. Since some numerical operations are involved, we import `numpy`. We also import the `denoise_wavelet()` function and `matplotlib.pyplot`.
+
+We then read the audio file using `wavfile.read()` function and normalize the amplitude as shown below:
+
+```python
+Fs, x = wavfile.read('guitar.wav') # Reading audio wave file
+x = x/max(x)   # Normalizing amplitude
+```
+
+We are reading the `guitar.wav` in the current directory. The signal samples will be stored in the `x` variable and the sampling frequency to `Fs` variable. 
+
+The amplitude is normalized because `wavfile` reads the audio in `int16` mode. This mode gives large values. Therefore, we need to reduce these values through normalization.
+
+Since the audio signal has no noise, we add noise similar to what we did for the ecg signal.
+
+```python
+sigma = 0.05  # Noise variance
+x_noisy = x + sigma * np.random.randn(x.size)   # Adding noise to signal
+
+# Wavelet denoising
+x_denoise = denoise_wavelet(x_noisy, method='VisuShrink', mode='soft', wavelet_levels=3, wavelet='sym8', rescale_sigma='True')
+```
+
+After adding the random noise, we denoise the signal using the `denoise_wavelet()` function. Here, we use the `VishuShrink` method. 
+For visualization, we plot the output.
+
+```python
+plt.figure(figsize=(20, 10), dpi=100)
+plt.plot(x_noisy)
+plt.plot(x_denoise)
+plt.show()
+```
+
+![Output for the denoised audio signal](/engineering-education/wavelet-transform-analysis-of-1d-signals-using-python/wavelet-three.png)
+
+The blue signal is the noisy signal, and the orange is the denoised output. Now, it is hard to know whether the denoising is done or not. It can be done by either calculating the peak signal to noise ratio(PSNR) or physically listening to the music. 
```

Comment:
```suggestion
The blue signal is the noisy signal, while the orange is the denoised output. It is hard to know whether the denoising is done or not. 

However, this can be done by either calculating the peak signal to noise ratio(PSNR) or physically listening to the music. 
```